tests/reftests/pin.test

```diff
@@ -1364,3 +1364,20 @@ echo GARBAGE>>"$1"
              error  2: File format error: unsupported or missing file format version; should be 2.0 or older
 [ERROR] No package named pin-at-future found.
 # Return code 5 #
+### :XI:g: git pin first retrieved as local
+### opam switch create vcs-local --empty
+### <pin:vcs-local/vcs-local.opam>
+opam-version: "2.0"
+### git -C vcs-local init -q --initial-branch=master
+### git -C vcs-local config core.autocrlf false
+### git -C vcs-local add vcs-local.opam
+### git -C vcs-local commit -qm 'init'
+### git -C vcs-local ls-files
+vcs-local.opam
+### <vcs-local/untracked>
+i shouldn't be retrieved
+### opam pin vcs-local ./vcs-local --no-action
+Package vcs-local does not exist, create as a NEW package? [y/n] y
+[vcs-local.dev] synchronised (file://${BASEDIR}/vcs-local)
+vcs-local is now pinned to git+file://${BASEDIR}/vcs-local#master (version dev)
+### test -f OPAM/vcs-local/.opam-switch/sources/vcs-local/untracked
```

Comment:
I would also modify `vcs-local/vcs-local.opam` after the git add so it has untracked changes and add a cat after the `test -f` to make sure the file is correctly copied